code/game/objects/structures/syndicate_uplink_beacon.dm

```diff
@@ -0,0 +1,115 @@
+			SCREENTIP_CONTEXT_RMB = "Deconstruct",
+		),
+	)
+	AddElement(/datum/element/contextual_screentip_tools, tool_behaviors)
+	listening_radio = new(src)
+	listening_radio.canhear_range = 0
+	teleport_beacon = new(src)
+
+/obj/structure/syndicate_uplink_beacon/attack_hand(mob/living/user, list/modifiers)
+	if(!user.mind?.has_antag_datum(/datum/antagonist/traitor))
+		balloon_alert(user, "don't know how to use!")
+		return
+	if(owner == WEAKREF(user))
+		balloon_alert(user, "already synchronized to you!")
+		return
+	if(owner != null)
+		balloon_alert(user, "already claimed!")
+		return
+	var/datum/looping_sound/typing/typing_sounds = new(src, TRUE)
+	balloon_alert(user, "beginning synchronization...")
+	if(!do_after(user = user, delay = 3 SECONDS, target = src, interaction_key = REF(src)))
+		balloon_alert(user, "interrupted!")
+		typing_sounds.stop()
+		return
+	typing_sounds.stop()
+	probe_traitor(user)
+
+/obj/structure/syndicate_uplink_beacon/screwdriver_act_secondary(mob/living/user, obj/item/tool)
+	tool.play_tool_sound(src)
+	balloon_alert(user, "deconstructing...")
+	if (!do_after(user, 5 SECONDS, target = src))
+		return FALSE
+	new /obj/item/stack/sheet/iron/five
+	new /obj/item/stack/cable_coil/five
+	teleport_beacon.forceMove(get_turf(src))
+	teleport_beacon = null
+	qdel(src)
+	return TRUE
+
+/obj/structure/syndicate_uplink_beacon/Destroy()
+	QDEL_NULL(listening_radio)
+	QDEL_NULL(teleport_beacon)
+	return ..()
+
+/// Proc reads the user, sets radio to the correct frequency and starts to listen for the replacement uplink code
+/obj/structure/syndicate_uplink_beacon/proc/probe_traitor(mob/living/user)
+	owner = WEAKREF(user)
+	var/datum/antagonist/traitor/traitor_datum = user.mind.has_antag_datum(/datum/antagonist/traitor)
+
+	uplink_code = traitor_datum.replacement_uplink_code
+	listening_radio.set_frequency(traitor_datum.replacement_uplink_frequency)
+	become_hearing_sensitive()
+
+/obj/structure/syndicate_uplink_beacon/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, list/message_mods, message_range)
+	if(ismob(speaker) || radio_freq != listening_radio.get_frequency())
+		return
+	if(!findtext(message, uplink_code))
+		return
+	teleport_uplink()
+
+/// Proc uses owners uplink handler to create replacement uplink and then lock or destroy their other uplinks
+/obj/structure/syndicate_uplink_beacon/proc/teleport_uplink()
+	if(!COOLDOWN_FINISHED(src, beacon_cooldown))
+		return
+	COOLDOWN_START(src, beacon_cooldown, 10 MINUTES)
+
+	var/mob/living/resolved_owner = owner.resolve()
+	if(!resolved_owner)
+		return
+
+	var/datum/antagonist/traitor/traitor_datum = resolved_owner.mind.has_antag_datum(/datum/antagonist/traitor)
+	if(!traitor_datum)
+		return
+
+	var/datum/uplink_handler/uplink_handler = traitor_datum.uplink_handler
+
+	SEND_SIGNAL(uplink_handler, COMSIG_UPLINK_HANDLER_REPLACEMENT_ORDERED)
+	new /obj/item/uplink(get_turf(src), resolved_owner, 0, uplink_handler)
```

Comment:
Can you make sure that this uplink is lockable in some way?

Reply:
> Can you make sure that this uplink is lockable in some way?

according to the PR, it’s not meant to be lockable. it’s supposed to be like a nukie/implant uplink which is always open.

Reply:
yes and I'm asking for it to be lockable

Reply:
> yes and I'm asking for it to be lockable

…why? why do you want something that obvious and easily destructible to have more security?

Reply:
because I don't want security to have access to an open uplink unless a traitor seriously fucks up and leaves it open for them.

Reply:
this is literally THE RESPONSE to having a unlocked uplink taken. it locks all lockable uplinks for the user and creates a disposable that they are supposed to secure and protect until they can get an implant or retrieve their old one. they can radiocode at any time to destroy it and create a new one at their beacon with the caveat that doing so alerts witnesses that they have said beacon.

Reply:
its lockable and is unlocked with your replacement code that you use to summon it, just if it hears the codeword and number combo

in the process of making it i found out that debug uplinks broke and i fixed that too.